File: src/main/java/com/tw/pos/discount/DiscountPromotion.java
package com.tw.pos.discount;


import com.tw.pos.entity.CartItem;

public class DiscountPromotion implements Discount {

    private double discountRate;

    public DiscountPromotion(double discountRate) {
        this.discountRate = discountRate / 100d;
    }

    @Override
    public double apply(CartItem cartItem, double price) {
        return cartItem.getNumber() * price * this.discountRate;
    }

    @Override
    public double getDiscountPrice(double originPrice) {
        return originPrice * discountRate;
    }
}

File: js/discount/discountPromotion.js

var DiscountPromotion = function(discounRate){
    this.discountRate = discounRate / 100;
};

DiscountPromotion.prototype.apply = function(cartItem, price){
    return cartItem.number * price * this.discountRate;
};

DiscountPromotion.prototype.getDiscountPrice = function(originPrice){
  return originPrice * this.discountRate;
};